src/components/UploadcareImage.tsx

```diff
@@ -1,7 +1,41 @@
 import Image, { ImageProps } from 'next/image';
 import React from 'react';
+import {
+  PLACEHOLDER_SIZE_FALLBACK,
+  PLACEHOLDER_SIZE_MULTIPLIER
+} from '../utils/constants';
+import { getInt } from '../utils/helpers';
 import { uploadcareLoader } from '../utils/loader';
 
+const shouldOverrideBlurDataUrl = (props: ImageProps): boolean => {
+  return props.placeholder === 'blur' && !props.blurDataURL;
+};
+
+const generateBlurDataUrl = (
+  src: string,
+  width: ImageProps['width']
+): string => {
+  const imageWidth = getInt(width);
+  const blurImageWidth = imageWidth
+    ? Math.ceil(imageWidth * PLACEHOLDER_SIZE_MULTIPLIER)
+    : PLACEHOLDER_SIZE_FALLBACK;
+  return uploadcareLoader({
+    src: src,
+    width: blurImageWidth,
+    quality: 1
+  });
+};
+
 export function UploadcareImage(props: ImageProps): JSX.Element {
-  return <Image loader={uploadcareLoader} {...props} />;
+  let blurDataURL: string | undefined;
+  if (typeof props.src === 'string' && shouldOverrideBlurDataUrl(props)) {
```

Comment:
Can we move the `typeof props.src === 'string'` condition into `shouldOverrideBlurDataUrl()` too?

Reply:
Actually yes, I would to. It will look like this:

```ts
  if (shouldOverrideBlurDataUrl(props)) {
    blurDataURL = generateBlurDataUrl(props.src as string, props.width);
  }
```

We need to use `as string` type assertion here because `shouldOverrideBlurDataUrl` isn't a type guard and TS won't calc `src`'s type properly. It's less obvious for me but probably more clean ✅